sparse/impl/KokkosSparse_sptrsv_solve_impl.hpp

```diff
@@ -121,91 +143,532 @@ struct SptrsvWrap {
           lhs(lhs_),
           rhs(rhs_),
           nodes_grouped_by_level(nodes_grouped_by_level_) {
-      KK_REQUIRE_MSG(!BlockEnabled, "Blocks are not yet supported.");
-      KK_REQUIRE_MSG(block_size_ == 0, "Blocks are not yet supported.");
+      KK_REQUIRE_MSG(block_size_ == 0,
+                     "Tried to use blocks with the unblocked Common?");
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    size_type get_block_size() const { return 0; }
+
+    // lset
+    KOKKOS_INLINE_FUNCTION
+    void lset(const size_type row, const scalar_t value) const {
+      lhs(row) = value;
+    }
+
+    // add. y += x
+    KOKKOS_INLINE_FUNCTION
+    static void add(const member_type &team, const scalar_t &x, scalar_t &y) {
+      Kokkos::single(Kokkos::PerTeam(team), [&]() { y += x; });
+      team.team_barrier();
```

Comment:
I think it would be better to move the ``team_barrier`` out of the function, even if it adds a few lines. It helps readability, since nothing about an ``add`` call indicates that it barriers unless you know this code well

Reply:
Looks like this add (and divide) were not even necessary for unblocked since add_and_divide is so easy with scalars.